### ISSUE TITLE
Support appending additional arguments

### ISSUE THREAD

**Issue by dizlexik** [frontsideair/yarnhook#30]
Would it be possible to add the ability to accept additional arguments so I could do something like this:

```
"husky": {
  "hooks": {
    "post-checkout": "yarnhook --ignore-optional",
    "post-merge": "yarnhook --ignore-optional",
    "post-rewrite": "yarnhook --ignore-optional"
  }
}
```

I'd really like to prevent yarnhook from installing optional dependencies.

**Comment by frontsideair** [comment]
I think adding `--ignore-optional` makes sense since they're optional and it'll make git operations faster.

I'm on the fence about adding arguments though, since—despite the name—yarnhook is package manager independent. I'm open to changing my mind but I'll make the `--ignore-optional` change regardless.

**Comment by zkochan** [comment]
I don't think this was a good idea. If someone wants to influence how the package manager is behaving itself, they can just create a config file for that package manager with settings. So if I want to skip optional dependencies, I would create a `.npmrc` with this content:

```
optional=false
```

Now because the CLI flags have the biggest priority, there is no way to include optional dependencies. I need optional dependencies to be installed. So now I have to always rerun `pnpm install` after switching branches. So yarnhook loses its benefits in my case.

**Comment by qnp** [comment]
@zkochan 100% agreed. In case of nuxt3, esbuild needs to be installed in the correct platform and is done using `optionalDependencies` so everytime yarnhook runs, the nuxt3 project cannot be built until a new `yarn install` is run.

**Comment by frontsideair** [comment]
I came back to this and I believe it was not the right call to exclude optional dependencies by default. While it's important that branch switches to be fast, it's more important that yarnhook doesn't change the end result. It should be a core principle that yarnhook doesn't install different dependencies than when you install them manually with default flags.

I reviewed the current default flags and this seems like the only offending one. If we categorize them:
- Preferring offline is fine, having internet connection shouldn't change behavior anyway
- Pure/frozen lockfile is good actually, it should be the default (and it is in many cases)
- No audit (npm only) is also good, it doesn't change the end result
- No save (also npm only) is also good, switching branches shouldn't change files (not sure if `npm install` does it, need to check)

So the first thing I'll do is to remove `--no-optional`, adding a way to configure it may come later. Not sure if it'll be needed as @zkochan demonstrated. I had a major refactor in a branch, I can probably add providing the option to that scope.

**Comment by frontsideair** [comment]
I cut a new release, it's a major one due to the reasons mentioned in the [release page](https://github.com/frontsideair/yarnhook/releases/tag/0.6.0). I hope this solves existing problems and please let me know if it breaks anything.
